pay-api/src/pay_api/models/eft_short_names.py

```diff
@@ -58,12 +58,22 @@ class EFTShortnames(Versioned, BaseModel):  # pylint: disable=too-many-instance-
     linked_by_name = db.Column('linked_by_name', db.String(100), nullable=True)
     linked_on = db.Column('linked_on', db.DateTime, nullable=True)
 
+    def generate_receipt_number(self, payment_id) -> str:
```

Comment:
You need to consider the CAS version suffix, also Short names wont need to be recreated as linked.

The receipt number is unique